azurerm/internal/services/authorization/resource_arm_role_definition.go

```diff
@@ -221,6 +221,15 @@ func resourceArmRoleDefinitionDelete(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
+	// fix bugs when scope is management group
+	// in this case, the id format is `/providers/Microsoft.Authorization/roleDefinitions/{id}`
+	// after func `parseRoleDefinitionId`, the scope is empty string
+	// read the real scope from state
+	if id.scope == "" {
+		scope := d.Get("scope").(string)
+		id.scope = strings.TrimPrefix(scope, "/")
+	}
```

Comment:
unfortunately this won't work where the resource has been removed from the Terraform Configuration - instead we'd be better to look up the Role Definition and confirm the level this Role Definition is created at - which should give us that information consistently?